Inlogg mot databas, hash+salt lösenord, globalt sparad inloggad användare

jkim931 opened the pull request:
Ny databas skapad (= ni måste köra klassen "DbLoginConfigUtil.java för att autogenerera ytterligare en configfil). Det finns nu två st. knappar i appen, en som loggar in snabbt med ett tryck och en som kollar i databasen med användarnamn och lösen. En ny klass körs som en application för att spara data globalt (t.ex. deviceID och userID för den inloggade).

jkim931 committed:
Databas för inloggning skapad + ny loginknapp

Appen kraschar just nu om databasen inte innehåller några användare, detta
arbetas på och kommer med i nästa commit. Utöver detta har klassen
LoginData skapats för att representera loginuppgifter i databasform,
dessutom underlättar klassen jämföring av uppgifter vid inloggning.
Hashning sker inte än, men kommer gå enkelt att lägga till senare. Utöver
detta finns nu även en hjälpklass som hanterar komplexa loginaktiviteter
så att vi slipper ha den koden i activity/fragmentklassen.

jkim931 committed:
Merge branch 'master' of https://github.com/Rovanion/AinAppen

jkim931 committed:
Login med lösen fixat!

Vi har nu två olika databaser, en för inloggningsuppgifter och en för
annan lokal data (cases, kontakter m.fl.)! Vi hashar inte lösen än utan
allt lagras i klartext, men själva jämförelsen kommer inte skilja sig
något.

jkim931 committed:
Fixade buggar vid login

Man kan nu lämna textfälten tomma och försöka logga in utan att det
kraschar! Dessutom så kan man gå in och ut ur appen och logga in utan att
det kraschar (gick inte tidigare)!

jkim931 committed:
Hashfunktion + salt för databas implementerat!

Ainappen lagrar inte längre lösenord i klartext! Nu finns istället det
hashade lösenordet och hashens SALT lagrat i databasen. Hashningen sköter
klassen "Hasher" och använder sig av SHA-256, dock så sker en del av det andra arbetet just nu i
"LoginActivity" vilket kanske inte är jättesnyggt, men jag vet inte hur vi
ska få bort koden därifrån på ett bra sätt!

jkim931 committed:
Inloggad user och device-id kan nu sparas globalt

Klassen "GlobalData" går nu att komma åt varsomhelst i appen och
försvinner inte om man sätter mobilen/appen i sleepmode. Dessutom så syns
det en liten toast direkt när man loggat in som bekräftar vilken användare
man är inloggad som!

eliaskarnsund commented:
För att kunna göra review:
1. Gör det som står med DbLoginConfigUtil.java
2. Högerklicka påDbLoginConfigUtil.java och klicka på Run as > Run Configurations 
3. Klicka på fliken classpath ta bort Android 4.2.2 (under bootstrap entries)
4. Ev. gör en clean på projektet (kanske steg 1 en gång till också)

eliaskarnsund commented on src/gov/polisen/ainappen/Hasher.java:
```diff
@@ -0,0 +1,36 @@
+	 */
+	public String getSHA256Hash(String input) {
+		MessageDigest md;
+		try {
```
Ska inte vara en try/catch, om algoritmen inte finns ska programmet crascha.

Klassen ska kasta ett exeption istället

Annars fungerar det!

Rovanion replied:
Eller att funktionen ska kasta undantaget men aja. Motiveringen är att den
här hashfunktionen måste existera eftersom det är på så sätt det är hashat
i databasen och vi måste vara konsekventa med samtliga lösenord i databasen.

2014-04-07 16:53 GMT+02:00 eliaskarnsund notifications@github.com:

> In src/gov/polisen/ainappen/Hasher.java:
> 
> > +/**
> > - \* Class used for hashing salt+password in AinAppen
> > - \* @author Joakim
> > - *
> > - */
> >   +public class Hasher {
> > -   /**
> > -    \* Method used to hash salt+password for AinAppen
> > -    \* returns the result in string format! Uses "ISO-8859-1" for 1-1 way formatting
> > -    \* purposes!
> > -    \* @param input
> > -    \* @return
> > -    */
> > -   public String getSHA256Hash(String input) {
> > -       MessageDigest md;
> > -       try {
> 
> Ska inte vara en try/catch, om algoritmen inte finns ska programmet
> crascha.
> 
> Klassen ska kasta ett exeption istället
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/Rovanion/AinAppen/pull/25/files#r11346868
> .

eliaskarnsund commented on src/gov/polisen/ainappen/LoginAcitivity.java:
```diff
@@ -7,42 +8,161 @@
+			});
+		}
+		public void makeFailedToast(){
+			String toastMessage = "Fel användarnamn eller lösenord!";
```
Borde vara en string i resources

samuelnilsson commented:
Koden fungerar bra och ser fin ut. Förstår inte exakt allt koden gör dock.

jkim931 committed:
Fixed several comments

Hasher now throws exception instead, said exception is handled with an
error message in the LoginActivity class notifying user a problem with a
lacking algorithm.

jkim931 committed:
Fixed a pseudo-random deviceID upon login

When logging in towards logindatabase a pseudorandom device ID is
generated.

jkim931 committed:
Merge branch 'master' into merge

Conflicts:
	res/values/strings.xml
	src/gov/polisen/ainappen/LocalDBHandler.java
	src/gov/polisen/ainappen/MainActivity.java

jkim931 committed:
Fixed for merge

Rovanion committed:
Merge pull request #19 from henninghall/master

Kamerafunktion och bildvisning (bevismaterial) för ärenden.

jkim931 committed:
Merge branch 'master' of github.com:j0wkin/AinAppen into merge

Conflicts:
	AndroidManifest.xml
	src/gov/polisen/ainappen/CaseListFragment.java
	src/gov/polisen/ainappen/MainActivity.java

jkim931 committed:
fixed a minor conflict
